src/main/java/org/jenkinsci/plugins/jenkinsreviewbot/ReviewboardPollingBuilder.java

```diff
@@ -108,56 +112,87 @@ public String getCheckBackPeriod() {
+    FileOutputStream fileOut = new FileOutputStream(file);
+    ObjectOutputStream out = new ObjectOutputStream(fileOut);
+    out.writeObject(reviewSet);
+    out.close();
+    fileOut.close();
   }
 
   @Override
   public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskListener listener) {
+    // Get file
+    String file = run.getParent().getRootDir() + tempFileName;
+    // Read from file
+    HashSet<Review.Slim> processedReviews = new HashSet();
+    listener.getLogger().println("Reading reviews from file...");
+    try {
+      processedReviews = readReviews(file);
+    } catch (Exception e) {
+      listener.getLogger().println("Couldn't read from file!");
+      e.printStackTrace(listener.getLogger());
+    }
+    listener.getLogger().println("Got " + processedReviews.size() + " reviews");
     listener.getLogger().println("Looking for reviews that need building...");
+    // Setup initial variables
     long period = checkBackPeriod != null && !checkBackPeriod.isEmpty() ? Long.parseLong(checkBackPeriod) : 1L;
     listener.getLogger().println("Going to check reviews updated during last " + period + " hour(s): ");
     ReviewboardConnection con = ReviewboardConnection.fromConfiguration();
+    // Get reviews from Reviewboard
+    listener.getLogger().println("Query: " + con.getPendingReviewsUrl(restrictByUser, reviewbotRepoId));
+    HashSet<Review.Slim> reviews = new HashSet();
     try {
-      listener.getLogger().println("Query: " + con.getPendingReviewsUrl(restrictByUser, reviewbotRepoId));
-      Collection<Review.Slim> reviews = ReviewboardOps.getInstance().getPendingReviews(con, period, restrictByUser, reviewbotRepoId);
-      listener.getLogger().println("Got " + reviews.size() + " reviews");
-      Set<Review.Slim> unprocessedReviews = new HashSet<Review.Slim>(reviews);
-      if (processedReviews != null) { //apparently, it is null when de-serialized from previous version of plugin... DUH!
-        Set<Review.Slim> richProcessed = getRichProcessedReviews();
-        unprocessedReviews.removeAll(richProcessed);
-      }
-      listener.getLogger().println("After removing previously processed, left with " + unprocessedReviews.size() + " reviews");
-      updateProcessed(reviews);
-      if (unprocessedReviews.isEmpty()) return;
+      reviews = new HashSet(ReviewboardOps.getInstance().getPendingReviews(con, period, restrictByUser, reviewbotRepoId));
+    } catch (Exception e) {
+      listener.getLogger().println("Couldn't get pending reviews from Reviewboard");
+      e.printStackTrace(listener.getLogger());
+    }
+    listener.getLogger().println("Got " + reviews.size() + " reviews");
+    // No need to continue if there are no reviews
+    if (reviews.isEmpty()) return;
+    // Write the reviews to the file, they will be processedReviews for the next run
+    listener.getLogger().println("Writing reviews to file...");
+    try {
+      writeReviews(file, reviews);
```

Comment:
Is this too early? If the job fails to start then this review will still be marked as processed and therefore never be run?